bom/internal/pom.xml

```diff
@@ -0,0 +1,305 @@
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-envers</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-realbackend</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-pojo-standalone-realbackend</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-showcase-library</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-v5migrationhelper-engine</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-v5migrationhelper-orm</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-coordination-outbox-polling</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-integrationtest-mapper-orm-batch-jsr352</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <!-- Jackson: used by the Elasticsearch REST client and in tests (wiremock, ...) -->
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-core</artifactId>
+                <version>${version.com.fasterxml.jackson}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-annotations</artifactId>
+                <version>${version.com.fasterxml.jackson}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-databind</artifactId>
+                <version>${version.com.fasterxml.jackson.databind}</version>
+            </dependency>
```

Comment:
If it's used by the Elasticsearch REST client, we need this in the public BOM.

Maybe we should have some automation to check that all non-test dependencies of public artifacts are managed in the BOM... because they must be.

However, the only way I can think to enforce that would be to [ban transitive dependencies](https://maven.apache.org/enforcer/enforcer-rules/banTransitiveDependencies.html) in public modules, which would force us to have explicit dependencies for everything in our public modules, which would in turn force us to have a version for every one of those dependencies. But damn, that's yucky. Definitely not something I'd like to see in Hibernate Search.

I'm open to ideas... :/

Reply:
After some back and forth, I've added these to the public BOM, and updated the comments next to libs/version
It looks like core is used by es client sniffer, while all three are used by avro and wiremock. Since avro is also in a list of public dependencies, I suppose we cannot use the trick to add these dependencies into specific modules using avro/wiremock directly without having them in the dependency management....

> Maybe we should have some automation to check that all non-test dependencies of public artifacts are managed in the BOM... because they must be.

There's this - https://maven.apache.org/enforcer/enforcer-api/writing-a-custom-rule.html I'll take a look if we could create some rule that would check for it, but that's probably for a different PR 😃

Reply:
> There's this - https://maven.apache.org/enforcer/enforcer-api/writing-a-custom-rule.html I'll take a look if we could create some rule that would check for it, but that's probably for a different PR smiley

Thanks! Agreed, this is out of scope.